### ISSUE TITLE
Multiple nested paths in functions within a function score query

### ISSUE THREAD

**Issue by Mahathi-Bhagavatula** [elastic/elasticsearch#21558]
I have a use case where the data is stored in multiple nested fields. As shown below:

```
{ "field1":
   { "sub_field1": "string",
     "sub_field2":"string"
   },
   { "sub_field1" :"string",
     "sub_field2":"string"
   }
},
{"field2":
   { "new_field1": "string",
     "new_field2":"string"
},
   { "new_field1" :"string",
     "new_field2":"string"
   }
}
```
Now, for each "query_str" I have to query all these nested fields and get the max (scores) w.r.t all these fields and I have to calculate the sum of all such query_str

This problem can be broken down into two sub problems :

1. For each Function Score Query, I can have different functions such that, each function should point out to each nested path. This will look some thing like below:

```
{ "query": {
  "function_score": {
  "functions": [ 
     { "query":{
        "nested": {
        "path": "field1",
          "query": {
          "filter": {
           "term": { "field1.sub_field1": "query_str" }}, 
        "weight": 1 }}}
     },
     { "query":{
        "nested": {
        "path": "field2",
           "query": {
           "filter": { 
             "term": { "field2.new_field1": "query_str" }}, 
           "weight": 1 }}}
      } ],
     "score_mode":"max"
     } 
   }
 }
```
2. Nested Score_modes:- after the above query, I will get some thing like max (scores) for one query string "query_str". Now, I need an addition of all such maximum scores for different query strings.
To summarize mathematically,
```
final_score =
      Sum on all queries ( for each_ query max ( score_on_field1, score_on_field2))
```

**Comment by cbuescher** [comment]
I think putting each of your queries in a boolean "should" clause should add the scores together in the end. That said, Elastic reserves GitHub for bug reports and feature requests but provides a [Discourse forum](https://discuss.elastic.co) and [IRC channels](https://www.elastic.co/community) for general questions, so I'm closing this. Feel free to reopen if it is clearer that this is something we cannot do currently.

**Comment by Mahathi-Bhagavatula** [comment]
@cbuescher I have added this in GitHub as I was referring to two new features in Elasticsearch. 
1) Now I cannot access two different nested paths for two different queries/ filters inside functions
2) In a way I cannot access two different nested paths at the same time. 
              {"nested":{"path":["field1","field2"].......
3) Score_mode can be "max" or "sum" for one particular set of queries. But I was referring to having a nested score_mode as sum (max)

So I am re-opening the issue and feel free to close in case these features are already present.

**Comment by clintongormley** [comment]
@Mahathi-Bhagavatula The suggestion by @cbuescher is the correct path to follow.  

> 1) Now I cannot access two different nested paths for two different queries/ filters inside functions
> 2) In a way I cannot access two different nested paths at the same time. 
{"nested":{"path":["field1","field2"].......

Yes, and we can't change that. A single `nested` query can only refer to a single `nested` field, it cannot cross fields as each nested field is a separate document. If you need access to two fields in the same context then they must be in the same document.
